### PR TITLE
feat(core): add support for pebble filters/functions as plugin

### PR DESCRIPTION
**Pull request by fhussonnois**
This PR is a POC to allow users to define custom filters and functions for pebble as plugins:

**Examples:** 

A custom Pebble Filter :

```java
public class UppercaseFilter extends PebbleFilter {

    @Override
    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self, EvaluationContext context, int lineNumber) throws PebbleException {
        return Optional.ofNullable(input)
            .map(Object::toString)
            .map(s -> s.toUpperCase(Locale.ROOT)).orElse(null);
    }

    @Override
    public List<String> getArgumentNames() {
        return List.of();
    }
}
```

A custom Pebble Function :

```java
public class UppercaseFunction extends PebbleFunction {

    @Override
    public Object execute(Map<String, Object> args, PebbleTemplate self, EvaluationContext context, int lineNumber) {
        if (!args.containsKey("value")) {
            throw new PebbleException(null, "The 'uppercase' function expects one arguments 'value'", lineNumber, self.getName());
        }
        String value = (String) args.get("value");
        return Optional.ofNullable(value).map(s -> s.toUpperCase(Locale.ROOT)).orElse(null);
    }

    @Override
    public List<String> getArgumentNames() {
        return List.of("value");
    }
}
```

**Limitations:**

 * This PR doesn't provide embedded plugin doc for Pebble functions/filters (i.e., those plugin are not visible in the UI).
 * This PR doesn't allow to provide custom unary/binary operators

### CODE REVIEW

File: core/src/main/java/io/kestra/core/expression/PebbleFilter.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    /**
+     * Returns the {@link Filter} name.
+     * 
+     * @return the string name  - cannot be {@link null}.
+     */
+    public String name() {
+        String className = getClass().getSimpleName()
+            .replace("Filter", "")
+            .replace(PebbleFilter.class.getSimpleName(), "");
```
Comment (brian-mulier-p) [inline core/src/main/java/io/kestra/core/expression/PebbleFilter.java]: If you trim "Filter" from the name you can't have "PebbleFilter" in the name afterwards :P

File: core/src/main/java/io/kestra/core/expression/PebbleFunction.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    /**
+     * Returns the {@link Function} name.
+     *
+     * @return the string name  - cannot be {@link null}.
+     */
+    public String name() {
+        String className = getClass().getSimpleName()
+            .replace("Function", "")
+            .replace(PebbleFunction.class.getSimpleName(), "");
```
Comment (brian-mulier-p) [inline core/src/main/java/io/kestra/core/expression/PebbleFunction.java]: Same as "Filter" trim

File: core/src/main/java/io/kestra/core/plugins/RegisteredPlugin.java [hunk 3]
```diff
@@ -85,6 +87,10 @@ public Class baseClass(String cls) {
             return TaskRunner.class;
         }
 
+        if (this.getPebbleExtensions().stream().anyMatch(r -> r.getName().equals(cls))) {
+            return TaskRunner.class;
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/plugins/RegisteredPlugin.java]: Seems not OK

**Review by anna-geller — APPROVED** [review]
nice!

File: core/src/main/java/io/kestra/core/expression/PebbleFilter.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+public abstract class PebbleFilter implements Filter, PebbleExtension {
+
+    /**
+     * Returns the {@link Filter} name.
+     * 
+     * @return the string name  - cannot be {@link null}.
+     */
+    public String name() {
+        String className = getClass().getSimpleName()
+            .replace(PebbleFilter.class.getSimpleName(), "")
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/expression/PebbleFilter.java:19]:
```suggestion
            .replace(this.getClass().getSimpleName(), "")
```

File: core/src/main/java/io/kestra/core/expression/PebbleFunction.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+public abstract class PebbleFunction implements Function, PebbleExtension {
+
+    /**
+     * Returns the {@link Function} name.
+     *
+     * @return the string name  - cannot be {@link null}.
+     */
+    public String name() {
+        String className = getClass().getSimpleName()
+            .replace(PebbleFunction.class.getSimpleName(), "")
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/expression/PebbleFunction.java:19]:
```suggestion
            .replace(this.getClass().getSimpleName(), "")
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**